lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -406,10 +406,10 @@ def main():
                     # if delta_time has been set, we require at least one
                     # neighbor within delta_time to accept a pixel in the image:
                     if delta_time is not None:
-                        cleaned_pixel_times = peak_time
-                        # makes sure only signal pixels are used in the time
+                        # make sure only signal pixels are used in the time
                         # check:
-                        cleaned_pixel_times[~signal_pixels] = np.nan
+                        cleaned_pixel_times = np.where(signal_pixels,
```

Comment:
Why are we modifying peak times in any case? All ctapipe functions should work with the cleaning mask instead of modifying the input arrays

Reply:
Indeed, that was the bug (which I introduced 4 years ago). I thought cleaned_pixel_times would be a copy of the original peak_times, and the nans would not be written out.

Reply:
But why is it even needed inside this function?

Reply:
The cleaning may use a timing constraint. The line in question is intended to ignore pixels which do not survive the rest of the (charge-based) cleaning conditions, when evaluating a pixel's time with respect to that of its neighbours (i.e. do not include in the constraint a neighbour which most likely contains only noise)

Reply:
But that's already covered by passing the mask of the previous step, right?

https://github.com/cta-observatory/ctapipe/blob/ec47070cd1691deafb952f5b4f51672891dde09b/src/ctapipe/image/cleaning.py#L281-L287

Reply:
Yes, I have just noticed that. Probably it was not the case when the cleaning was  implemented within lstchain.